### ISSUE TITLE
Translation persist in global memory, overriding defaults

### ISSUE THREAD

**Issue by 23b** [PrestaShop/PrestaShop#24277]
#### Describe the bug

After loading a translation, the default language string will not be returned correctly.
This only happens if you access multiple languages in a single request.

#### Expected behavior

Receive original string. 

#### Steps to Reproduce

1. Consider a module with a translation file in translation/de.php but no translation/en.php 
`$m = Module::getInstanceByName('psgdpr');`
2. We load a translated string, the result will be stored in the global $MODULES
`$de = $m->l('Message','exportdatatocsv','de-DE');`
3. Now we want to get the default string
`$en = $m->l('Message','exportdatatocsv','en-US');`
4. Erroneously we receive the German translation stored in $MODULES  at this point
`echo 'de: '.$de.PHP_EOL.'en: '.$en;`

#### Additional information

* PrestaShop version: 1.7.7.3
* PHP version: 1.7.2
fixed by #24189

**Comment by florine2623** [comment]
Hello @23b ,

Do you have a module example so we can test this ?

Thanks!

**Comment by 23b** [comment]
example:
```php
$m = Module::getInstanceByName('psgdpr');
$de = $m->l('Message','exportdatatocsv','de-DE');
$en = $m->l('Message','exportdatatocsv','en-US');
echo 'de: '.$de.PHP_EOL.'en: '.$en;
```

result:
```
de: Nachricht
en: Nachricht
```

**Comment by florine2623** [comment]
Do you have the zip version of the module ?

**Comment by 23b** [comment]
> Do you have the zip version of the module ?

https://github.com/PrestaShop/psgdpr

**Comment by hibatallahAouadni** [comment]
Hello @23b 

We are aware of this issue, it is already in our debug backlog. This issue is a duplicate of #13392. 
To be informed when it's fixed, please subscribe to the issue mentioned above.

PS: I'm going to change the **Fixed ticket** in your PR to this one #13392

Thanks!

**Comment by hibatallahAouadni** [comment]
Duplicate of #13392
